app/controllers/items_controller.rb

```diff
@@ -19,14 +30,29 @@ def create
         else
             render 'new'
         end
+    
+        respond_to do |format|
+            format.html # show.html.erb
+            format.json { render :json => {:item => @item } }
+        end
```

Comment:
Render and/or redirect will be called multiple times when the method body executed.